core/services/ocr2/plugins/ocr2keeper/evm21/logprovider/provider_life_cycle.go

```diff
@@ -19,6 +19,7 @@ var (
 )
 
 func (p *logEventProvider) RefreshActiveUpkeeps(ids ...*big.Int) ([]*big.Int, error) {
+	// TODO investigate how we can batch the refresh
```

Comment:
nit. can we call it exploratory? e.g. https://github.com/smartcontractkit/chainlink/blob/develop/core/services/ocr2/plugins/ocr2keeper/evm21/logprovider/provider_life_cycle.go#L65 

We are trying to differentiate b/w TODOs that we want to do before launch and exploratory optimisations we can do later